From nobody@cs.rice.edu  Fri Jun 19 09:28:41 1998
Received: from web.cs.rice.edu (web.cs.rice.edu [128.42.1.163])
          by cs.rice.edu (8.8.5/8.8.4) with ESMTP
	  id JAA22045 for <plt-gnats@cs.rice.edu>; Fri, 19 Jun 1998 09:28:41 -0500 (CDT)
Message-Id: <199806191428.JAA17053@web.cs.rice.edu>
Date: Fri, 19 Jun 1998 09:28:40 -0500 (CDT)
From: dhavlik@rice.edu
Reply-To: dhavlik@rice.edu
To: plt-gnats@cs.rice.edu
Subject: 'internal-error' received when I place a #<define> in a with-handlers statement 

>Number:         653
>Category:       mcmicmac
>Synopsis:       'internal-error' received when I place a #<define> in a with-handlers statement
>Confidential:   no
>Severity:       non-critical
>Priority:       low
>Responsible:    shriram
>State:          closed
>Class:          sw-bug
>Submitter-Id:   unknown
>Arrival-Date:   Sat Jun 20 16:24:28 GMT 1998
>Last-Modified:  Thu Jul  9 20:38:23 CDT 1998
>Originator:     Doug Havlik
>Organization:
titan
>Release:        52
>Environment:
SunOS4
>Description:
The following code produces the following error message:
(with-handlers
    (((lambda (exn) #f)
      (lambda (exn) (void))))
  (define x 1)
  )

internal-error.report-error: #(struct:internal-definition (#(struct:symbol #(struct:origin source source) #(struct:location 3 11 79 #<object>) #(struct:location 3 11 79 #<object>) x x ())) #(struct:number #(struct:origin source source) #(struct:location 3 13 81 #<object>) #(struct:location 3 13 81 #<object>) 1)): report error.Internal error: aries:annotate: unknown object to annotate, #(struct:internal-definition (#(struct:symbol #(struct:origin source source) #(struct:location 3 11 79 #<object>) #(struct:location 3 11 79 #<object>) x x ())) #(struct:number #(struct:origin source source) #(struct:location 3 13 81 #<object>) #(struct:location 3 13 81 #<object>) 1))
>How-To-Repeat:
enter above code
>Fix:

>Audit-Trail:

Responsible-Changed-From-To: steck->shriram
Responsible-Changed-By: shriram
Responsible-Changed-When: Thu Jul  9 19:48:42 1998
Responsible-Changed-Why:

State-Changed-From-To: open-closed
State-Changed-By: shriram
State-Changed-When: Thu Jul  9 20:36:34 1998
State-Changed-Why:
Fixed the bug.  The problem occurred when there was only a single
internal definition in a begin context.  Then parse-expr "optimized"
and expanded the expression directly, instead of wrapping it in begin, 
so the top-level-handling code of begin failed, exposing this error.

>Unformatted:
